scripts/windows/UninstallWindowsAgent.ps1

```diff
@@ -0,0 +1,95 @@
+# Function to check if Powershell is running as Administrator
+function Test-Administrator {
+    $currentUser = [Security.Principal.WindowsIdentity]::GetCurrent()
+    $principal = New-Object Security.Principal.WindowsPrincipal($currentUser)
+    return $principal.IsInRole([Security.Principal.WindowsBuiltInRole]::Administrator)
+}
+
+if (-not (Test-Administrator)) {
+    Write-Warning "This script needs to be run as an administrator."
+    exit
+}
+
+# Create HKCR Mapping
+New-PSDrive -Name "HKCR" -PSProvider Registry -Root "HKEY_CLASSES_ROOT" -ErrorAction SilentlyContinue
+
+# Function to recursively search for DisplayName in the registry
+function Find-JumpCloudGUID {
+    $rootKey = "HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\Installer\UserData\S-1-5-18\Products"
+    $uninstallKey = "HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall"
+    $jumpcloudGUIDs = @()
+
+    # Searching Installed Products
+    Get-ChildItem -Path $rootKey | ForEach-Object {
+        $installPropertiesPath = "$($_.PSPath)\InstallProperties"
+        try {
+            $displayName = (Get-ItemProperty -Path $installPropertiesPath -ErrorAction SilentlyContinue).DisplayName
+            if ($displayName -like "*JumpCloud Agent*") {
+                $guid = $_.PSChildName
+                $jumpcloudGUIDs += $guid
+            }
+        } catch {
+            Write-Host "Error accessing $installPropertiesPath"
+        }
+    }
+
+    # Searching for the Uninstall Key
+    Get-ChildItem -Path $uninstallKey | ForEach-Object {
+        try {
+            $displayName = (Get-ItemProperty -Path $_.PSPath -ErrorAction SilentlyContinue).DisplayName
+            if ($displayName -like "*JumpCloud Agent*") {
+                $uninstallGUID = $_.PSChildName
+                $jumpcloudGUIDs += $uninstallGUID
+            }
+        } catch {
+            Write-Host "Error accessing $($_.PSPath)"
+        }
+    }
+
+    return $jumpcloudGUIDs
+}
+
+# Function to remove registry keys, JC folder, and JC service
+function Remove-JumpCloud {
+    $guids = Find-JumpCloudGUID
+
+    foreach ($guid in $guids) {
```

Comment:
I got GUID and MSI Install ID when I ran `Find-JumpCloudGUID`. Both should not be looped together.